gadget/gadget.go

```diff
@@ -273,15 +283,15 @@ func systemOrSnapID(s string) bool {
 	return true
 }
 
-// ReadInfo reads the gadget specific metadata from gadget.yaml
-// in the snap. classic set to true means classic rules apply,
-// i.e. content/presence of gadget.yaml is fully optional.
-func ReadInfo(gadgetSnapRootDir string, classic bool) (*Info, error) {
+// ReadInfo reads the gadget specific metadata from gadget.yaml in the snap. If
+// constraints is nil, ReadInfo will just check for self-consistency, otherwise
+// rules for the classic or system seed cases are enforced.
+func ReadInfo(gadgetSnapRootDir string, constraints *ModelConstraints) (*Info, error) {
```

Comment:
`ModelConstraints` is small enough that I think we can allow it to be passed by value. Also, this allows to drop all further checks for nil. So I'd suggest to use:
```
func ReadInfo(gadgetSnapRootDir string, constraints ModelConstraints) (*Info, error) 
```

Reply:
the idea was for nil to mean just check for consistency